### PR TITLE
Fix SSL handshake for a secondary connection

### LINKED ISSUE #3059: SDL is unable to start protected Audio/Video service in multi-transport configuration
### Bug Report
Within https://github.com/smartdevicelink/sdl_core/pull/2164 it was introduced a possibility for SDL to handle multiple transports. E.g.: RPC service could utilize BT connection whereas Audio/Video services use WIFI.
Currently SDL is unable to start secure Audio/Video service in such configuration even if RPC service in protected mode has been started successfully.

##### Reproduction Steps
1. Set up SDL for secure sessions:
   - set SDL certificate  
   - set Mobile root and issuing CA certificates
2. Set up SDL for multiple transports:
   - MultipleTransportsEnabled = true
   - SecondaryTransportForBluetooth = WiFi
   - AudioServiceTransports = TCP_WIFI
   - VideoServiceTransports = TCP_WIFI
3. Set force protection for Audio/Video services:
   - ForceProtectedService = 0x0A, 0x0B
4. Make sure SDL and Android mobile device connected to the same WIFI and BT
5. Start SDL, Web-HMI
6. In SPT select `IPC MP` connection type (which means multiple transports)
7. Register new NAVI app
8. Switch RPC service to protected mode (this step is optional; it's added only to be sure secure connection can be established for RPC service)
9. Activate app on Web-HMI
10. Try to start secure Video streaming

##### Expected Behavior
SDL starts secure streaming successfully

##### Observed Behavior
SDL responds with NACK to mobile app

##### OS & Version Information
* OS/Version: Ubuntu 18.04 amd64
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/develop (66888d4)
* Testing Against: FordSPT, Web-HMI

##### Test Case, Sample Code, and / or Example App
 - [SmartDeviceLinkCore_NACK_encrypted.log.tar.gz](https://github.com/smartdevicelink/sdl_core/files/3677830/SmartDeviceLinkCore_NACK_encrypted.log.tar.gz)

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Fixes #3059 

This PR is **not ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Tested manually

### Summary
There was found a several places in SDL code where SSL handshake was working wrong when secured service was started from a secondary connection like WIFI. All these places were updated to consider primary connection as a source for an SSL context and other attributes for case when the current connection is secondary.
Looks like this is a part which was not considered during initial implementation of the secondary transport feature.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 1]
```diff
@@ -1124,6 +1124,26 @@ const uint8_t ConnectionHandlerImpl::GetSessionIdFromSecondaryTransport(
   return 0;
 }
 
+Connection* ConnectionHandlerImpl::GetPrimaryConnection(
+    const ConnectionHandle connection_handle) const {
+  LOG4CXX_DEBUG(logger_,
+                "Getting primary connection for ID " << connection_handle);
+  ConnectionList::const_iterator it = connection_list_.find(connection_handle);
+  if (connection_list_.end() == it) {
+    LOG4CXX_ERROR(
```
[the remark this answers is not in the document]
Comment (AKalinich-Luxoft) [reply src/components/connection_handler/src/connection_handler_impl.cc]: @iCollin agree particularly.
I removed error logging from `SetSSLContext` \ `GetSSLContext` \ `SetProtectionFlag` \ `GetHandshakeContext` as it really looks superfluous, however I left the only error logging in `GetPrimaryConnection` as this is the only way to figure out from SDL logs what went wrong if connection was dropped in ConnectionHandler class.

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 2]
```diff
@@ -1675,13 +1703,14 @@ bool ConnectionHandlerImpl::IsHeartBeatSupported(
     uint8_t session_id) const {
   LOG4CXX_AUTO_TRACE(logger_);
   sync_primitives::AutoReadLock lock(connection_list_lock_);
-  uint32_t connection = static_cast<uint32_t>(connection_handle);
-  ConnectionList::const_iterator it = connection_list_.find(connection);
-  if (connection_list_.end() == it) {
-    LOG4CXX_WARN(logger_, "Connection not found !");
+  uint32_t connection_id = static_cast<uint32_t>(connection_handle);
```
Comment (AByzhynar) [inline src/components/connection_handler/src/connection_handler_impl.cc]: @AKalinich-Luxoft `const`?

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 3]
```diff
@@ -1124,6 +1124,26 @@ const uint8_t ConnectionHandlerImpl::GetSessionIdFromSecondaryTransport(
   return 0;
 }
 
+Connection* ConnectionHandlerImpl::GetPrimaryConnection(
+    const ConnectionHandle connection_handle) const {
+  LOG4CXX_DEBUG(logger_,
+                "Getting primary connection for ID " << connection_handle);
+  ConnectionList::const_iterator it = connection_list_.find(connection_handle);
+  if (connection_list_.end() == it) {
```
Comment (AByzhynar) [inline src/components/connection_handler/src/connection_handler_impl.cc]: @AKalinich-Luxoft can we use `helpers::in_range` to make more clear & short code?

File: src/components/connection_handler/src/connection_handler_impl.cc (hunk 2 above)
Comment (AKalinich-Luxoft) [reply src/components/connection_handler/src/connection_handler_impl.cc]: @AByzhynar fixed in https://github.com/smartdevicelink/sdl_core/pull/3065/commits/5676d3e72f8cf438604136ce4c85efb66e408a38

File: src/components/connection_handler/src/connection_handler_impl.cc (hunk 3 above)
Comment (AKalinich-Luxoft) [reply src/components/connection_handler/src/connection_handler_impl.cc]: @AByzhynar unfortunately, we need to have iterator here to get the pointer to appropriate connection in the code below, so we have to use `find` in this scope

**Comment by AKalinich-Luxoft** [comment]
@theresalech PR is ready for Livio team review

**Pull request merged**